src/lib/zf/zf.c

```diff
@@ -36,6 +42,13 @@ int zf_init(void)
     /* Failure here is non-fatal. */
   }
 
+ if( attr->log_file && strcmp(zf_log_file_name, attr->log_file) != 0 )
+    strcpy(zf_log_file_name, attr->log_file);
+ else if( attr->log_to_kmsg )
+    strcpy(zf_log_file_name, "/dev/kmsg");
+ else if( attr->log_file == NULL && zf_log_file_name[0] == '\0' )
+    strcpy(zf_log_file_name, "stderr");
```

Comment:
I had a few comments about this block, so I'll merge them into a single comment as they somewhat depend on each other.

I think `attr->log_file` might also need to be handled carefully here to avoid overflows. Alternatively, can we drop this entire block? I think `zf_log_file_name` should be appropriately set by the above `zf_log_redirect`.

Should the `attr->log_to_kmsg` check come first here to match the above block, or is this the desired behaviour?

Should `"stderr"` here, be `"/dev/stderr"` instead? If we do drop this block, this should probably be the value we assign `zf_log_file_name` when we define it.

Also, if we do keep this block, I think the indentation of the `if`/`else if` lines are slightly off.

Reply:
thanks, yes I think it is a good idea to drop most of the block. `zf_log_redirect` sets `zf_log_file_name` properly and the initial value of `zf_log_file_name` can be `/dev/stderr` when it is defined. I can't see how I can get rid of `attr->log_to_kmsg` case, which in turn made me realize `zf_log_replace_stderr` missedd setting `zf_log_file_name`.